src/screens/CreateTokenScreen.tsx

```diff
@@ -0,0 +1,132 @@
+import theme from '../lib/theme'
+import { Typography } from '../components/Typography'
+import { useSecretSelector } from '../hooks/use-secret-selector'
+
+const styles = StyleSheet.create({
+  container: {
+    paddingHorizontal: theme.spacing(3),
+    paddingTop: theme.spacing(4),
+  },
+  description: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(4),
+  },
+  button: {
+    marginTop: theme.spacing(1),
+    justifyContent: 'center',
+  },
+})
+
+type Props = NativeStackScreenProps<MainStackParamList, 'CreateToken'>
+
+export const CreateTokenScreen = ({ route, navigation }: Props) => {
+  const { secretId } = route.params
+  const secret = useSecretSelector(secretId)
+  const { user } = useAuth()
+  const [subscriptionId, setSubscriptionId] = useState<string>('')
+  const [description, setDescription] = useState('')
+  const { update } = useSecrets()
+  const expoToken = usePushToken()
+
+  const api = useMemo(() => apiFactory({ idToken: user.idToken }), [user])
+
+  const disabled = description.length < 3
+
+  const isFocused = useIsFocused()
+  const ref = useRef(null)
+
+  useEffect(() => {
+    if (isFocused) {
+      ref.current && ref.current.focus()
+    }
+  }, [isFocused])
+
+  const handleGenerateToken = async () => {
+    try {
+      const token = await api.generateToken(secret, subscriptionId)
+      const newToken = {
+        token,
+        description,
+      }
+      const existingTokens = secret.tokens ? secret.tokens : []
+      const secretUpdated = {
+        ...secret,
+        tokens: [newToken, ...existingTokens],
+      }
+
+      await update(secretUpdated)
+
+      navigation.replace('Token', {
+        secretId,
+        token,
+      })
+      Toast.show('Token successfully created')
+    } catch (err) {
+      Toast.show('An error occurred generating the token')
+      console.error(err)
+    }
+  }
+
+  useEffect(() => {
+    if (!user || !expoToken) return
+
+    const register = async () => {
+      const id = await api.registerSubscription({
+        type: 'expo',
+        token: expoToken,
+      })
+      setSubscriptionId(id)
+    }
+
+    register()
+  }, [user, api, expoToken])
+
+  if (!secret) {
+    return null
+  }
+
+  return (
+    <View style={styles.container}>
+      <View>
+        <Typography variant="h6">
+          Insert a description for this token
+        </Typography>
+      </View>
+      <View style={styles.description}>
+        <TextInput
+          ref={ref}
+          textAlign="left"
+          label="Description"
+          accessibilityLabel="Description"
+          placeholder="Description"
+          placeholderTextColor={theme.colors.disabled}
+          mode="outlined"
+          value={description}
+          onChangeText={setDescription}
+          multiline
+        />
+      </View>
+      <Button
+        style={styles.button}
+        icon="plus"
+        mode="contained"
+        onPress={handleGenerateToken}
+        disabled={disabled}
```

Comment:
nitpick: The button is not disabled while waiting for the server answer. You can spam the button to generate a few tokens in a row.